src/lib/openjp2/jp2.c

```diff
@@ -2636,6 +2658,104 @@ static OPJ_BOOL opj_jp2_read_ftyp(opj_jp2_t *jp2,
     jp2->jp2_state |= JP2_STATE_FILE_TYPE;
 
     return OPJ_TRUE;
+}
+
+static OPJ_BOOL opj_jp2_read_asoc(	opj_jp2_t *jp2,
+									OPJ_BYTE * p_header_data,
+									OPJ_UINT32 p_header_size,
+									opj_event_mgr_t * p_manager)
+{
+    OPJ_UINT32 label_tag;
+    OPJ_UINT32 asoc_tag;
+    OPJ_UINT32 asoc_size;
+    opj_jp2_asoc_t *asoc;
+
+    /* preconditions */
+    assert(jp2 != 00);
+    assert(p_header_data != 00);
+    assert(p_manager != 00);
+

```

Comment:
you should check p_header_size is at least 8 bytes long here

Reply:
Fixed